topic-operator/src/main/java/io/strimzi/operator/topic/v2/TopicEvent.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic.v2;
+
+import io.strimzi.api.kafka.model.KafkaTopic;
+
+import java.util.Objects;
+
+/**
+ *
+ * Represents a change in Kube relating to a KafkaTopic
+ */
+sealed interface TopicEvent permits TopicUpsert, TopicDelete {
+    long ageNs();
+    String namespace();
+    String name();
+    String resourceVersion();
+
+    default KubeRef toRef() {
+        return new KubeRef(namespace(), name(), 0);
+    }
+}
+/**
+ * The creation or update of a KafkaTopic.
+ * Note that this may include the change which adds a metadata.deletionTimestamp.
+ */
+record TopicUpsert(long nanos, String namespace, String name, String resourceVersion) implements TopicEvent {
```

Comment:
is `nanos` the "timestamp" of the event? Can we make it clearer with a better name?

Reply:
It results from a call to `System.nanoTime()`. Therefore it's a value that has no meaning on it's own (the epoch it not defined). It can only be used with the result of another `System.nanoTime()`, e.g. to compute a duration. So it's a bit tricky to know _what_ to call it. Do you have any suggestions?

Reply:
> It results from a call to System.nanoTime(). Therefore it's a value that has no meaning on it's own

well should it have a meaning? Why this nanoTime() call is made, which is the purpose? I see you are using to process an age ... `System.nanoTime() - nanos` ... but age of what? Of the `TopicUpsert` itself? So is the nanos parameter a kind of born date?

Reply:
> Why this nanoTime() call is made, which is the purpose?

That's a good catch. We're not currently using it really (apart from the `toString()` output that isn't really called). I was using it to record the total time it took to reconcile an event. We could remove it, but we will be adding a metric (`strimzi.reconciliations.duration`) which will need it, so I'm inclined to keep it.

> So is the nanos parameter a kind of born date?

Yes, a kind of born "date", but relative to an unknown epoch (in particular it's not a milliseconds offset from the unix epoch,  like from `System.currentTimeMillis()`). So call it `bornDate` would be misleading, I think. I suppose `nanosOffset` would be sort-of better.  Wdyt?

Reply:
>  I suppose nanosOffset would be sort-of better. Wdyt?

Well the parameter would be a `nanosStartOffset` because you are going to get the "elapsed" nanos time via `System.nanoTime() - nanosStartOffset` where the current nanos time would be the end offset.